pkg/asset/installconfig/azure/validation.go

```diff
@@ -312,10 +335,22 @@ func validateInstanceTypes(client API, ic *types.InstallConfig) field.ErrorList
 			}
 			ultraSSDEnabled := strings.EqualFold(ultraSSDCapability, "Enabled")
 			allErrs = append(allErrs, ValidateInstanceType(client, fieldPath.Child("platform", "azure"),
-				ic.Azure.Region, instanceType, diskType, computeReq, ultraSSDEnabled, vmNetworkingType, zones)...)
+				ic.Azure.Region, instanceType, diskType, computeReq, ultraSSDEnabled, vmNetworkingType, zones, architecture)...)
 		}
 	}
 
+	// we are going to use the defaultInstanceType so let's check it. The assumption here is that since cp and compute arches cannot differ today, it's ok to
+	// not check the default instance as long as it is used in any one place.
+	if !useDefaultInstanceType && defaultInstanceType != "" {
```

Comment:
Your comment says "we are going to use the defaultInstanceType". So shouldn't this be:
```suggestion
	if useDefaultInstanceType && defaultInstanceType != "" {
```

Reply:
no, i poorly worded it . what i  meant is - we are checking to see if defaultinstancetype has been provided but not been used above, in which case we need to check it specifically to see if it matches the correct architecture. i will change the wording.

Reply:
updated the comment.